src/crawlee/static_content_crawler/_static_content_crawler.py

```diff
@@ -0,0 +1,166 @@
+    of that it implements the HTTP communication using the HTTP clients. The class allows integration with
+    any HTTP client that implements the `BaseHttpClient` interface. The HTTP client is provided to the crawler
+    as an input parameter to the constructor.
+    StaticContentCrawler is generic class and is expected to be used together with specific parser that will be used to
+    parse http response and type of expected TCrawlingContext which is available to the user function.
+    See prepared specific version of it: BeautifulSoupCrawler, ParselCrawler or HttpCrawler for example.
+
+    The HTTP client-based crawlers are ideal for websites that do not require JavaScript execution. However,
+    if you need to execute client-side JavaScript, consider using a browser-based crawler like the `PlaywrightCrawler`.
+    """
+
+    def __init__(
+        self,
+        *,
+        parser: StaticContentParser[TParseResult],
+        additional_http_error_status_codes: Iterable[int] = (),
+        ignore_http_error_status_codes: Iterable[int] = (),
+        **kwargs: Unpack[BasicCrawlerOptions[TCrawlingContext]],
+    ) -> None:
+        self._parser = parser
+
+        kwargs.setdefault(
+            'http_client',
+            HttpxHttpClient(
+                additional_http_error_status_codes=additional_http_error_status_codes,
+                ignore_http_error_status_codes=ignore_http_error_status_codes,
+            ),
+        )
+
+        if '_context_pipeline' not in kwargs:
+            raise ValueError(
+                'Please pass in a `_context_pipeline`. You should use the '
+                'StaticContentCrawler._create_static_content_crawler_pipeline() method to initialize it.'
+            )
+
+        kwargs.setdefault('_logger', logging.getLogger(__name__))
+        super().__init__(**kwargs)
+
+    def _create_static_content_crawler_pipeline(self) -> ContextPipeline[ParsedHttpCrawlingContext[TParseResult]]:
+        """Create static content crawler context pipeline with expected pipeline steps."""
+        return (
+            ContextPipeline()
+            .compose(self._make_http_request)
+            .compose(self._parse_http_response)
+            .compose(self._handle_blocked_request)
+        )
+
+    async def _parse_http_response(
+        self, context: HttpCrawlingContext
+    ) -> AsyncGenerator[ParsedHttpCrawlingContext[TParseResult], None]:
+        parsed_content = await self._parser.parse(context.http_response)
+        yield ParsedHttpCrawlingContext.from_http_crawling_context(
+            context=context,
+            parsed_content=parsed_content,
+            enqueue_links=self._create_enqueue_links_function(context, parsed_content),
+        )
+
+    def _create_enqueue_links_function(
+        self, context: HttpCrawlingContext, parsed_content: TParseResult
+    ) -> EnqueueLinksFunction:
+        async def enqueue_links(
```

Comment:
Nice :+1:.

I understand that this is a private method, but could we add some docstrings here? Since we consider writing your own crawlers a legitimate use case, it should be understandable for users who want to do so.

This applies to the other private methods here as well.

Reply:
Done